### PR TITLE
LPS-159026 Localizable button is able in System Fields side panel details

### PR DESCRIPTION
**Pull request by daiazius**
Related issue: https://issues.liferay.com/browse/LPS-159026

### CODE REVIEW

**Comment by daiazius** [comment]
ci:test:sf

**Comment by daiazius** [comment]
ci:test:relevant

**Comment by daiazius** [comment]
ci:test:sf

**Comment by daiazius** [comment]
ci:test:relevant

**Comment by daiazius** [comment]
ci:test:relevant

**Comment by daiazius** [comment]
ci:test:relevant

**Comment by daiazius** [comment]
ci:test:relevant

**Comment by daiazius** [comment]
ci:test:relevant

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/EditObjectField.tsx [hunk 1]
```diff
@@ -479,6 +479,16 @@ export default function EditObjectField({
...
+	useEffect(() => {
+		if (values.system && objectName !== 'AccountEntry') {
+			const localizationButton = document.querySelector(
+				'.dropdown-toggle'
+			);
+
+			localizationButton?.setAttribute('disabled', 'true');
+		}
+	}, [objectName, values.system]);
+
```
Comment (danvb112) [inline modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/EditObjectField.tsx:491]: Hey @daiazius, this behavior of just disabling the input and not the flags is default of the [Clay Localized Input component](https://clayui.com/docs/components/localized-input.html), so I'm not sure if disabling it via JavaScript is the best solution. I think we have to send a message to the frontend infrastructure team that is responsible for Clay, and we ask if they can develop this functionality for us, and then we just use it. If for some reason they are unable to develop this feature in this component for us, we can proceed with their solution. Please let me know if you have any questions and update me via Slack on their decision. :wink:

**Review by danvb112 — CHANGES_REQUESTED** [review]

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/EditObjectField.tsx [hunk 2]
```diff
@@ -479,6 +479,16 @@ export default function EditObjectField({
...
 	}, [objectFields]);
 
+	useEffect(() => {
+		if (values.system && objectName !== 'AccountEntry') {
+			const localizationButton = document.querySelector(
+				'.dropdown-toggle'
+			);
+
+			localizationButton?.setAttribute('disabled', 'true');
+		}
```
Comment (danvb112) [inline modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/EditObjectField.tsx:489]: Hey @daiazius, I think we can move this logic into an existing `useEffect` in the `InputLocalized` component, and create a new property for the component, like `disableFlags` with `boolean` type, so the useEffect inside the `InputLocalized` component will be like :

```tsx
useEffect(() => {
    if (disableFlag) {
	const localizationButton = document.querySelector(
		'.dropdown-toggle'
	);

	localizationButton?.setAttribute('disabled', 'true');
    }

    const locale =
	availableLocales.find(({label}) => label === selectedLocale)! ??
	availableLocales[0];
    setLocale(locale);
}, [disableFlag, selectedLocale]);

```

And in `EditObjectField.tsx` we can pass the new property to `InputLocalized` as:

```tsx
disableFlag={values.system && objectName !== 'AccountEntry'}
```

**Review by danvb112 — CHANGES_REQUESTED** [review]

**Comment by daiazius** [comment]
ci:test:relevant

**Comment by daiazius** [comment]
ci:test:sf

**Comment by daiazius** [comment]
ci:test:relevant

**Review by danvb112 — APPROVED** [review]
LGTM! :tada:

**Comment by danvb112** [comment]
ci:forward

**Comment by danvb112** [comment]
ci:forward

**Comment by daiazius** [comment]
ci:test:relevant

**Comment by danvb112** [comment]
ci:forward

**Pull request closed without merging**